src/main/java/mate/academy/springboot/swagger/service/impl/SortServiceImpl.java

```diff
@@ -0,0 +1,33 @@
+package mate.academy.springboot.swagger.service.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import mate.academy.springboot.swagger.service.SortService;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+
+@Service
+public class SortServiceImpl implements SortService {
+    @Override
+    public List<Sort.Order> parseSortingCondition(String sortCondition) {
+        List<Sort.Order> orders = new ArrayList<>();
+        if (sortCondition.contains(":")) {
+            String[] sortingFields = sortCondition.split(";");
+            for (String field : sortingFields) {
+                Sort.Order order;
+                if (field.contains(":")) {
+                    String[] fieldsAndDirections = field.split(":");
+                    order = new Sort.Order(Sort.Direction.valueOf(fieldsAndDirections[1]),
+                            fieldsAndDirections[0]);
+                } else {
+                    order = new Sort.Order(Sort.Direction.DESC, field);
+                }
+                orders.add(order);
+            }
+        } else {
+            Sort.Order order = new Sort.Order(Sort.Direction.DESC, sortCondition);
+            orders.add(order);
+        }
+        return orders;
+    }
```

Comment:
20 lines it is not so bad, but maybe we can move some logic to the separate private methods to increase the readability

Reply:
And also each magic value is better to make constant, especially ":" that you use twice. You can name it DELIMITER for example